integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -672,4 +560,19 @@ class CarbonFileMetastore extends CarbonMetaStore {
       case _ => throw new NoSuchTableException(tableIdentifier.database.get, tableIdentifier.table)
     }
   }
+
+  def removeStaleTimeStampEntries(sparkSession: SparkSession): Unit = {
+    val tablesList = sparkSession.sessionState.catalog.listDatabases().flatMap {
+      database =>
+        sparkSession.sessionState.catalog.listTables(database)
+          .map(table => s"${ database }_${ table }")
+    }
+    val invalidTableIds = CarbonMetadata.getInstance().getAllTables.asScala.collect {
+      case carbonTable if !tablesList.contains(carbonTable.getTableUniqueName) =>
```

Comment:
Make sure the table uniqueName is case insensitive

Reply:
done